.github/workflows/deploy.backend-api.yml

```diff
@@ -0,0 +1,116 @@
+  # Publish Db Migrations 
+  publish_DbMigrations:
+    name: Publish Database Migrations Bundle
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Setup .NET Core
+        uses: actions/setup-dotnet@v3
+        with:
+          dotnet-version: ${{ env.DOTNET_VERSION }}
+
+      - name: Restore repo dotnet tools
+        run: dotnet tool restore
+        shell: bash
+
+      - name: Build bundle
+        run: dotnet ef migrations bundle -p ${{ env.WEB_APP_PROJECT }} -o ${{ github.workspace }}/efbundle -r linux-x64 --self-contained --configuration ${{ env.DOTNET_BUILD_CONFIG }}
+        shell: bash
+
+      - name: Publish migrations bundle artifact
+        uses: actions/upload-artifact@v4
+        with:
+          path: ${{ github.workspace }}
+          name: migrations
+
+  # Publish Web App
+  publish_webapp:
+    name: Publish Web App
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Get Git Commit Hash
+        id: vars
+        run: |
+          calculatedSha=$(git rev-parse --short ${{ github.sha }})
+          echo "GitHash=$calculatedSha" >> $GITHUB_ENV
+
+      - name: Setup .NET Core
+        uses: actions/setup-dotnet@v3
+        with:
+          dotnet-version: ${{ env.DOTNET_VERSION }}
+      
+      - name: dotnet publish
+        run: dotnet publish ${{ env.WEB_APP_PROJECT }} -c ${{ env.DOTNET_BUILD_CONFIG }} -o ${{ github.workspace }} -p:GitHash=${{ env.GitHash }}
+        shell: bash
+
+      - name: Publish webapp artifact
+        uses: actions/upload-artifact@v4
+        with:
+          path: ${{ github.workspace }}
+          name: webapp
+
+  # Deploy UAT
+  deploy-uat:
+    name: Deploy UAT
+    needs: [publish_DbMigrations, publish_webapp]
+    runs-on: ubuntu-latest
+    environment: uat
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Setup .NET Core
+        uses: actions/setup-dotnet@v3
+        with:
+          dotnet-version: ${{ env.DOTNET_VERSION }}
+
+      - name: Set Git Permissions
+        run: chmod -R +rwx .git
+
+      - name: Download artifact from build job
+        uses: actions/download-artifact@v4
+        with:
+          path: migrations
+          name: migrations
+
+      - name: Run Migrations
+        run: |
+          chmod +x ./efbundle
+          ./efbundle --connection "${{ secrets.UAT_DB_CONNECTION_STRING }}"
+        working-directory: ${{ github.workspace }}/migrations
+        shell: bash
+
+      - name: Download artifact from build job
+        uses: actions/download-artifact@v4
+        with:
+          path: webapp
+          name: webapp
+           
+      - name: 'Run Azure webapp deploy action using publish profile credentials'
+        uses: azure/webapps-deploy@v2
+        with: 
+          app-name: ${{ env.AZURE_WEBAPP_BACKEND_NAME }}
+          publish-profile: ${{ secrets.AZURE_WEBAPP_BACKEND_PUBLISH_PROFILE  }}
+          package: ${{ env.AZURE_WEBAPP_PACKAGE_PATH }}
```

Comment:
You want to publish the webapp artifact that you download in the previous step here - similar to the migrations, and frontend. Currently you'll be publishing the project code, not the build artifact.